### DISCUSSION TITLE
Calling native function in IChatCompletionService

### DISCUSSION THREAD

**Discussion by imranshams** [microsoft/semantic-kernel#7001] [Q&A] [answered]
I'm trying to develop an agent which write and send an email for me based on [this link](https://learn.microsoft.com/en-us/semantic-kernel/concepts/agents?pivots=programming-language-csharp) 

I want to integrate semantic kernel in my asp.net razor page. It's a chat page which will support the user with writing and sending email.

I have injected my plugin with native function and this is the result of **DebugPlugins** function.
`
**********************************************
****** Registered plugins and functions ******
**********************************************

Plugin: EmailPlugin
   send_email: Sends an email to a recipient.
      Params:
      - recipientEmail:
        default: ''
      - subject:
        default: ''
      - body:
        default: ''`

So, on the UI side, everything is going well until the last step **(6:	Send the email to the boss)**. 
When the user asked the system to send the email, it says:

_I'm sorry, but I don't actually have the capability to send emails. However, you can easily copy the draft below and send it through your email client._

I spend a lot of time to find related sample, but all of existing samples in repo are based on console app or notebooks. In real world scenario we need web app integration as well as console app.

Code:
[Code.zip](https://github.com/user-attachments/files/16033130/Code.zip)

**Comment by stephentoub** [comment, marked as answer]
Try changing this:
```C#
await foreach (var item in _aiChatService.GetStreamingChatMessageContentsAsync(aiHistory, executionSettings: executionSettings))
```
to this:
```C#
await foreach (var item in _aiChatService.GetStreamingChatMessageContentsAsync(aiHistory, executionSettings: executionSettings, _kernel))
```
The kernel object is what contains the plugins. Those need to be provided to the service.

**Reply by imranshams** [reply]
That was exactly the point, I couldn't find it in any documentation. Thank you so much for your time.
